### PR TITLE
chore: remove python and wasm_python executors

### PR DESCRIPTION
**Pull request by frrist**
As a part of the work required to land https://github.com/bacalhau-project/bacalhau/issues/2351 this PR removes the `JobSpecLanguage`, its executor, and the wasm_python executor. https://github.com/bacalhau-project/bacalhau/pull/2381 follows this (is based on) with the rest of the changes for #2351

### CODE REVIEW

**Review by simonwo — APPROVED** [review]
Love a good spring clean!

**Comment by frrist** [comment]
@lukemarsden I will await an approval from you before enabling auto-merge.

**Comment by lukemarsden** [comment]
Go ahead!

**Pull request merged**
